src/main/resources/templates/allAlbums.html

```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>All the songs</title>
+</head>
+<body>
+<h1> All the Albums</h1>
+
+<div th:each="album : ${albums}">
+<h3 th:text="${album.title}">Album Title</h3>
+<p>
+    Artist: <span th:text="${album.artist}"></span>
+
+</p>
+    <p>
+        Number of singles: <span th:text="${album.songCount}"></span>
+    </p>
+    <p>
+        Length:<span th:text="${album.length}"></span>
+    </p>
+    <a th:href="${album.imageUrl}">Click here</a>
```

Comment:
Instead of using a anchor tag and linking to the image, why not sure an img tag and show the image?